Update tailwind.md to use v4 setup

offblck opened the pull request:
Updating the tailwind setup guide to use the new 4.0 version.
- no need for `tailwind.config.js` anymore
- in included CSS file only need `@import "tailwindcss";`
- cli package and command changed to `@tailwindcss/cli`

Some heads up even if this doesn't get merged:
- In the previous version the numbering of steps was reset by code snippets, fixed it by indenting

jkelleyrtp commented:
I'd like to point out that both v3 and v4 exist - I think it's fine to promote v4 as the "recommend" way of using tailwind with dioxus, but our template still uses v3 so these docs will go out of date. 

Maybe two sections - one for v3 and one for v4 would be better until we've completely switched over?

jkelleyrtp reviewed (changes requested):
I'd like to point out that both v3 and v4 exist - I think it's fine to promote v4 as the "recommend" way of using tailwind with dioxus, but our template still uses v3 so these docs will go out of date.

Maybe two sections - one for v3 and one for v4 would be better until we've completely switched over?

offblck commented:
Thank you for the feedback! I will create two sections, one for `Setup with Tailwind v4 (Recommended)` which comes first and one for `Setup with Tailwind v3` if that sounds good. And I guess it should be noted that the templates still use v3.

offblck commented:
Committed a version containing guides for both versions.
- noted under the v3 header that dioxus cli currently creates templates with that version
- changed the v3 tailwind cli link to point the correct version (they changed it)
- added both cli commands for development server

Let me know if I should change something around
